### PR TITLE
adding redirects for s.dvc.org

### LINKED ISSUE #2967: doc: shorter links to user-guide
Hi, I need a few shortlinks for the following, which I need for `dvc exp init`.

1. https://dvc.org/doc/user-guide/project-structure/pipelines-files
2. https://dvc.org/doc/user-guide/experiment-management/running-experiments

I'd prefer something like `https://dvc.org/doc/pipelines-files` or even `https://dvc.org/pipelines-files`. Or, something else that is much shorter and still readable/understandable.

Here's how we display it right now in `dvc exp init`:
[asciicast](https://asciinema.org/a/RU9HNkdfuq40zKvSsAOgYTygg)

Looking at the dvc's code, we only seem to have short links for `man.dvc.org` which redirects to command-reference. We could also have something similar like `https://guide.dvc.org/running-experiments` (though it's 2 characters lengthier than `https://dvc.org/doc/running-experiments`).

### PR DESCRIPTION
**Pull request by iesahin**
> Per https://github.com/iterative/dvc.org/issues/2967#issuecomment-956455055
"for now to just use redirects-list entries with some convention that the new solution will follow?"

adds the following redirects for s.dvc.org

`s.dvc.org/g/exp/run` -> `/doc/user-guide/experiment-management/running-experiments`
`s.dvc.org/g/exp` -> `/doc/user-guide/experiment-management`
`s.dvc.org/g/` -> `/doc/user-guide/`
`s.dvc.org/s/` -> `/doc/start/`
`s.dvc.org/c/` -> `/doc/use-cases/`
`s.dvc.org/r/` -> `/doc/command-reference`

Closes #2968
Closes #2967

### CODE REVIEW

File: redirects-list.json [hunk 1]
```diff
@@ -17,6 +17,13 @@
   "^https://error\\.dvc\\.org/(.*)?                                                       https://dvc.org/doc/user-guide/troubleshooting#$1 303",
   "^https://(code|data|remote)\\.dvc\\.org/(.+)                                           https://s3-us-east-2.amazonaws.com/dvc-public/$1/$2 303",
 
+  "^https://s\\.dvc\\.org/g/exp/run(.*)?$                                                https://dvc.org/doc/user-guide/experiment-management/running-experiments$1",
+  "^https://s\\.dvc\\.org/g/exp/(.*)?$                                                   https://dvc.org/doc/user-guide/experiment-management/$1",
```
Comment (casperdcl) [inline redirects-list.json:21]:
```suggestion
  "^https://s\\.dvc\\.org/g/exp/(.*)?$                                                   https://dvc.org/doc/user-guide/experiment-management/$1",
  "^https://s\\.dvc\\.org/g/pipeline-files/(.*)?$                                        https://dvc.org/doc/user-guide/project-structure/pipelines-files/$1",
```

really closes #2968

**Review by casperdcl — CHANGES_REQUESTED** [review]

**Comment by jorgeorpinel** [comment]
> Closes #2967

I don't think this closes https://github.com/iterative/dvc.org/issues/2967#issuecomment-956455055

File: redirects-list.json [hunk 2]
```diff
@@ -17,6 +17,14 @@
...
+  "^https://s\\.dvc\\.org/g/pipeline-files/(.*)?$                                        https://dvc.org/doc/user-guide/project-structure/pipelines-files/$1",
+  "^https://s\\.dvc\\.org/g/(.*)?$                                                       https://dvc.org/doc/user-guide/$1",
+  "^https://s\\.dvc\\.org/s/(.*)?$                                                       https://dvc.org/doc/start/$1",
+  "^https://s\\.dvc\\.org/c/(.*)?$                                                       https://dvc.org/doc/use-cases/$1",
+  "^https://s\\.dvc\\.org/r/(.*)?$                                                       https://dvc.org/doc/command-reference/$1",
```
Comment (jorgeorpinel) [inline redirects-list.json:26]: If these are temporary, we should use `301` I think.

Comment (shcheklein) [reply redirects-list.json:26]: I think it should be the same as we do for the short command reference links?

Comment (iesahin) [reply redirects-list.json:26]: I don't think these are temporary, but not permanent for HTTP 301. 

A general URL shortener solution seems another site to manage and I believe this one provides a good solution we already use. 

So, an HTTP 302 may be used as a status code.

Comment (shcheklein) [reply redirects-list.json:26]: 302 (temporary) sounds right to me... probably @jorgeorpinel also meant 302

**Review by skshetry — APPROVED** [review]
Personally, I don't like URLs with these shortcodes. This will likely make users pause for a second before they open them, at least I would do that.

Instead of having `dvc.org/{s,c,r,g}/`, we could manually have unique short links in the front-matter (which dvc.org engine could verify for uniqueness) as suggested by @shcheklein in https://github.com/iterative/dvc.org/pull/2968#discussion_r736585596. That'd definitely require adding to each doc entry, which I think is fine for UI purposes.

**Comment by shcheklein** [comment]
> That would probably take another month to implement in the engine.

front-matter yes, removing prefixes (and use some custom link) - it can be done even today. frontmatter is just an implementation detail, I don't thing @skshetry is specifically concerned about the mechanism

**Comment by skshetry** [comment]
> That would probably take another month to implement in the engine.

I could have waited if this was going to be worked on in the next month. 😄  I had a PR that only added 2 links: #2968, but I did not get any response (that adding those redirections is okay for now).

**Comment by shcheklein** [comment]
@skshetry please feel free to add any redirects under the `s.dvc.org` namespace with the redirects.json file. I think that should resolve the problem for now (if let's say you still don't like the verbosity of the Emre's redirects)?

**Pull request merged**